internal/search/graphql_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package search
+
+import (
+	"github.com/go-openapi/swag"
+	. "github.com/onsi/ginkgo/v2/dsl/core"
+	. "github.com/onsi/ginkgo/v2/dsl/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/oran-o2ims/internal/model"
+)
+
+var _ = Describe("GraphQL filter", func() {
+	var selectorParser *SelectorParser
+	BeforeEach(func() {
+		var err error
+		selectorParser, err = NewSelectorParser().
+			SetLogger(logger).
+			Build()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	DescribeTable(
+		"Map a filter string to a SearchFilter list (Cluster)",
+		func(filter string, expected []*model.SearchFilter) {
+			actual, err := GraphqlFilter(filter).MapFilter(selectorParser, func(s string) string {
+				return GraphqlPropertyCluster(s).MapProperty()
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(actual).To(Equal(expected))
+		},
+		Entry(
+			"Single term",
+			"(eq,resourcePoolID,spoke0)",
+			[]*model.SearchFilter{
+				{
+					Property: "cluster",
+					Values:   []*string{swag.String("=spoke0")},
```

Comment:
In other tests we are using the [k8s.io/utils/ptr](https://pkg.go.dev/k8s.io/utils@v0.0.0-20231127182322-b307cd553661/ptr). It is not very different, but has the advantage of generics, so you can use the same `ptr.To` function for all types:

```go
Values: []*string{ptr.To("=spoke0")},
```